qcodes_contrib_drivers/drivers/Keysight/Keysight_E36313A.py

```diff
@@ -0,0 +1,111 @@
+                           label="Channel {} Voltage".format(chan),
+                           get_cmd='SOURCE{:d}:VOLT?'.format(chan),
+                           get_parser=float,
+                           set_cmd='SOURCE{:d}:VOLT {{:.8G}}'.format(chan),
+                           unit='V')
+
+        self.add_parameter('source_current',
+                           label="Channel {} Current".format(chan),
+                           get_cmd='SOURCE{:d}:CURR?'.format(chan),
+                           get_parser=float,
+                           set_cmd='SOURCE{:d}:CURR {{:.8G}}'.format(chan),
+                           unit='A')
+
+        self.add_parameter('voltage',
+                           get_cmd='MEAS:VOLT? (@{:d})'.format(chan),
+                           get_parser=float,
+                           label='Channel {} Voltage'.format(chan),
+                           unit='V')
+
+        self.add_parameter('current',
+                           get_cmd='MEAS:CURR? (@{:d})'.format(chan),
+                           get_parser=float,
+                           label='Channel {} Current'.format(chan),
+                           unit='A')
+
+        self.add_parameter('resistance',
+                           get_cmd='MEAS:RES? (@{:d})'.format(chan),
+                           get_parser=float,
+                           label='Channel {} Resistance'.format(chan),
+                           unit='ohm')
+
+        self.add_parameter('voltage_limit',
+                           get_cmd='SENS{:d}:VOLT:PROT?'.format(chan),
+                           get_parser=float,
+                           set_cmd='SENS{:d}:VOLT:PROT {{:.8G}}'.format(chan),
+                           label='Channel {} Voltage Limit'.format(chan),
+                           unit='V')
+
+        self.add_parameter('current_limit',
+                           get_cmd='SENS{:d}:CURR:PROT?'.format(chan),
+                           get_parser=float,
+                           set_cmd='SENS{:d}:CURR:PROT {{:.8G}}'.format(chan),
+                           label='Channel {} Current Limit',
+                           unit='A')
+
+        self.add_parameter('enable',
+                           get_cmd='OUTP{:d}?'.format(chan),
+                           set_cmd='OUTP{:d} {{:d}}'.format(chan),
+                           val_mapping={'on':  1, 'off': 0})
+
+        self.add_parameter('source_mode',
+                           get_cmd=':SOUR{:d}:FUNC:MODE?'.format(chan),
+                           set_cmd=':SOUR{:d}:FUNC:MODE {{:s}}'.format(chan),
+                           val_mapping={'current': 'CURR', 'voltage': 'VOLT'})
+
+        self.channel = chan
+
+
+class E36313A(VisaInstrument):
+    """
+    This is the qcodes driver for the Keysight E36313A programmable DC power supply
+
+    Status: 
+    TODO:
+        - 
+    """
+    def __init__(self, name, address, **kwargs):
+        super().__init__(name, address, terminator='\n', **kwargs)
```

Comment:
if not done as below, users will not be able to change the terminator if they happen to need to.
```suggestion
    def __init__(self, name, address, terminator='\n', **kwargs):
        super().__init__(name, address, terminator=terminator, **kwargs)
```